### PR TITLE
Refactor components, don't crash gulp on error

### PR DESCRIPTION
**Pull request by luqmaan**
I noticed a lot of the data is actually missing. So a good park to test
is Republic Square. Example: https://i.imgur.com/2RJQF6Y.png
- Don't crash gulp on error
- Add utils/api.js for making api calls
- Simplify render methods
- Add propTypes to components
- Remove jquery

### CODE REVIEW

File: client/js/components/App.react.jsx [hunk 1]
```diff
@@ -1,71 +1,63 @@
...
-        getParksJson().then((data) => {
-            var parkByID = {}
-            data.forEach((current, index) => {
-                // NOTE: Coordinates are lng lat here
-                current.center.reverse();
-                parkByID[current.park_id] = current;
-            });
-            this.setState({parkList: data, parkByID: parkByID});
-        });
+        this.state = {
```
Comment (aaronmgdr) [inline client/js/components/App.react.jsx:12]: i believe setting this.state is highly discouraged/ will break react. define `this.getInitialState()` or  call `this.setState()`

File: client/js/components/ParkList.jsx [hunk 2]
```diff
@@ -24,3 +22,7 @@ export default class ParkList extends React.Component {
     }
 }
 
+ParkList.propTypes = {
+    parks:  React.PropTypes.array.isRequired,
```
Comment (aaronmgdr) [inline client/js/components/ParkList.jsx:26]: are'nt propTypes the best?

File: client/js/components/App.react.jsx (hunk 1 above)
Comment (luqmaan) [inline client/js/components/App.react.jsx:12]: No. This is the way you do it with ES6.

Comment (luqmaan) [inline client/js/components/App.react.jsx:12]: http://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html#es6-classes

Comment (aaronmgdr) [inline client/js/components/App.react.jsx:12]: oh never mind then. 

good to know.

File: client/js/components/ParkList.jsx (hunk 2 above)
Comment (johntyree) [inline client/js/components/ParkList.jsx:26]: `s/propTypes/types/`

**Comment by luqmaan** [comment]
:deciduous_tree:

**Pull request merged**